camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/DslDeferredFunctionCall.java

```diff
@@ -0,0 +1,215 @@
+                method = Reflections.getMethodFromArgs(instance, fnName, instanceArgs);
+            }
+        }
+
+        if (method.isPresent()) {
+            Method m = method.get();
+
+            checkCallAllowed(m);
+
+            try {
+                // Value is most likely another BrooklynDslDeferredSupplier - let the caller handle it,
+                return Maybe.of(Reflections.invokeMethodFromArgs(instance, m, instanceArgs));
+            } catch (IllegalArgumentException | IllegalAccessException | InvocationTargetException e) {
+                // If the method is there but not executable for whatever reason fail with a fatal error, don't return an absent.
+                throw Exceptions.propagate(new InvocationTargetException(e, "Error invoking '"+fnName+"("+toString(instanceArgs)+")' on '"+instance+"'"));
+            }
+        } else {
+            return Maybe.absent(new IllegalArgumentException("No such function '"+fnName+"("+toString(args)+")' on "+obj));
+        }
+    }
+
+    protected Maybe<?> resolve(Object object, boolean immediate) {
+        if (object instanceof DslFunctionSource || object == null) {
+            return Maybe.of(object);
+        }
+
+        Maybe<?> resultMaybe = Tasks.resolving(object, Object.class)
+                .context(((EntityInternal)entity()).getExecutionContext())
+                .deep(true)
+                .immediately(immediate)
+                .recursive(false)
+                .getMaybe();
+
+        if (resultMaybe.isPresent()) {
+            // No nice way to figure out whether the object is deferred. Try to resolve it
+            // until it matches the input value as a poor man's replacement.
+            Object result = resultMaybe.get();
+            if (result == object) {
+                return resultMaybe;
+            } else {
+                return resolve(result, immediate);
+            }
+        } else {
+            return resultMaybe;
+        }
+    }
+
+    private static void checkCallAllowed(Method m) {
+        DslAccessible dslAccessible = m.getAnnotation(DslAccessible.class);
+        boolean isAnnotationAllowed = dslAccessible != null;
+        if (isAnnotationAllowed) return;
+
+        // TODO white-list using brooklyn.properties (at runtime)
+
+        Class<?> clazz = m.getDeclaringClass();
+        Package whiteListPackage = BrooklynDslCommon.class.getPackage();
+        boolean isPackageAllowed = (clazz.getPackage() != null && // Proxy objects don't have a package
+                    clazz.getPackage().getName().startsWith(whiteListPackage.getName()));
+        if (isPackageAllowed) {
+            if (DEPRECATED_ACCESS_WARNINGS.add(m)) {
+                log.warn("Deprecated since 0.11.0. The method '" + m.toString() + "' called by DSL should be white listed using the " + DslAccessible.class.getSimpleName() + " annotation. Support for DSL callable methods under the " + whiteListPackage + " will be fremoved in a future release.");
```

Comment:
typo "fremoved".   Would be nice to break this line up over several (currently 297 characters wide!)